fix: clarify how to access individual attrs from fetchUserAttributes (#2988)
aws-amplify/docs: docs/lib/auth/fragments/android/user_attributes/10_fetch_attributes.md

> ```java

Reviewer: Is there an RxJava example for this API, too? Could do like:
```java
Amplify.Auth.fetchUserAttributes()
    .filter(attribute -> AuthUserAttributesKey.email().equals(attribute.getKey()))
    .subscribe(email -> Log.i("AuthDemo", "User email = " + email.getValue()));
```

```kotlin
Amplify.Auth.fetchUserAttributes(

>     { attributes ->
>         Log.i("AuthDemo", "All user attributes = $attributes")
> 
>         // To access a specific user attribute:
>         val attributesMap: MutableMap<AuthUserAttributeKey, String> = HashMap()
>         for (attribute in attributes) {
>             attributesMap[attribute.key] = attribute.value
>         }
>         val email = attributesMap[AuthUserAttributeKey.email()]
>         if (email != null) {
>             Log.i("AuthDemo", "User email = $email")
>         }
>     }, 

Reviewer: More closely aligned to what @royjit suggested for Swift, how about:

```suggestion
    { attributes ->
        Log.i("AuthDemo", "All user attributes = $attributes")

        // To access a specific user attribute:
        val email = attributes
            .firstOrNull { it.key == AuthUserAttributeKey.email() }?.value
        if (email != null) {
            Log.i("AuthDemo", "User email = $email")
        }
    }, 
```
